ext/ffi/lib.rs

```diff
@@ -184,6 +184,32 @@ enum NativeType {
   F32,
   F64,
   Pointer,
+  Struct(Vec<NativeType>),
+}
+
+impl NativeType {
+  fn get_size(&self) -> usize {
+    match self {
+      NativeType::Void => 0,
+      NativeType::U8 | NativeType::I8 => 1,
+      NativeType::U16 | NativeType::I16 => 2,
+      NativeType::U32 | NativeType::I32 | NativeType::F32 => 4,
+      NativeType::U64
+      | NativeType::I64
+      | NativeType::F64
+      | NativeType::Pointer => 8,
+      NativeType::USize | NativeType::ISize => std::mem::size_of::<usize>(),
```

Comment:
This already makes an assumption that size of pointer is 8, which is also the size of `usize`. Maybe merge `Pointer` match with `size_of::<usize>`
```suggestion
      NativeType::USize | NativeType::Pointer | NativeType::ISize => std::mem::size_of::<usize>(),
```

Reply:
Sure, will do. Though currently the pointers API just assumes 64 bit pointers as well 😆